libraries/AP_RangeFinder/AP_RangeFinder_Benewake_TFMiniPlus.cpp

```diff
@@ -149,20 +150,31 @@ void AP_RangeFinder_Benewake_TFMiniPlus::update()
     }
 }
 
-bool AP_RangeFinder_Benewake_TFMiniPlus::process_raw_measure(le16_t distance_raw, le16_t strength_raw,
+void AP_RangeFinder_Benewake_TFMiniPlus::process_raw_measure(le16_t distance_raw, le16_t strength_raw,
                                                              uint16_t &output_distance_cm)
 {
     uint16_t strength = le16toh(strength_raw);
+    const uint16_t MAX_DIST_CM = 1200;
+    const uint16_t MIN_DIST_CM = 10;
 
     output_distance_cm = le16toh(distance_raw);
 
-    if (strength < 100 || strength == 0xFFFF) {
-        return false;
+    if (strength < 100 || strength == 0xFFFF || output_distance_cm > MAX_DIST_CM) {
+        /*
+         * From manual: "when the signal strength is lower than 100 or equal to
+         * 65535, the detection is unreliable, TFmini Plus will set distance
+         * value to 0." - force it to the max distance so status is set to OutOfRangeHigh
+         * rather than NoData.
+         */
+        output_distance_cm = MAX_DIST_CM;
+    } else {
+        output_distance_cm = constrain_int16(output_distance_cm, MIN_DIST_CM, MAX_DIST_CM);
     }
 
-    output_distance_cm = constrain_int16(output_distance_cm, 10, 1200);
-
-    return true;
+    /* Respect maximum from parameter  */
+    if (output_distance_cm == MAX_DIST_CM) {
```

Comment:
The maximum effective sensor value of 12 meters is treated as an invalid value.

Reply:
That is what the other sensors are doing. See libraries/AP_RangeFinder/AP_RangeFinder_Benewake.cpp

```cpp
reading_cm = MAX(model_dist_max_cm(), max_distance_cm() + BENEWAKE_OUT_OF_RANGE_ADD_CM);
```

So if the param is configured to exactly 12m, then 12m will be handled as an invalid value as we will actually return 13m and make update_status() set it to "out of range". It's ill-defined if the used set it to more than what the sensor is capable of. By separating this logic we could make it handle the case no parameter value is set, a way for the user to tell "maximum is whatever the sensor supports").

IMO this should be changed after the changes mentioned above.